docs/src/index.md

```diff
@@ -268,6 +268,228 @@ df2 = @eachrow df begin
+function wrapper with the syntax `@byrow`. 
+
+`@byrow` is not a "real" macro and cannot be used outside of 
+DataFramesMeta.jl macros, however it's behavior within DataFramesMeta.jl
+macros should be indistinguishable from externally defined macros. 
+Thought of as a macro `@byrow` accepts a single argument and 
+creates an anonymous function wrapped in `ByRow`.  For example,
+
+```julia
+@transform(df, y = @byrow :x == 1 ? "true" : "false)
+```
+
+becomes
+
+```julia
+transform(df, :x => ByRow(x -> x == 1 ? "true", "false") => :y)
+```
+
+Macros that accept `@byrow`:
+
+* `@transform` and `@transform!`, `@select`, `@select!`, and `@combine`. 
+  `@byrow` appears in the right hand side of expressions, of the form 
+  `@select(df, z = @byrow :x * :y)`. 
+* `@where` and `@orderby`, with syntax of the form `@where(df, @byrow :x > :y)`
+* `@with`, where the anonymous function created by `@with` is wrapped in
+  `ByRow`. `@with(df, @byrow :x * :y)`, which is conceptually similar to 
+  `map((x, y) -> x * y, df.x, df.y)`.
+
+To avoid writing `@byrow` multiple times when performing multiple
+operations, it is allowed to use`@byrow` at the beginning of a block of 
+operations. All transformations in the block will operate by row.
+
+```julia
+julia> @where df @byrow begin 
+           :a > 1
+           :b < 5
+       end
+1×2 DataFrame
+ Row │ a      b     
+     │ Int64  Int64 
+─────┼──────────────
+   1 │     2      4
+```
+
+`@byrow` can be used on macros which accept `GroupedDataFrame`s,
+however, like with `ByRow` in DataFrames.jl, when `@byrow` is
+used, functions do not take advantage of the grouping, so for
+example the result of `@transform(df, y = @byrow f(:x))` and 
+`@transform(groupby(df, :g), y = @byrow f(:x))` is the same.
+
+### Comparison with `@eachrow`
+
+In previous versions of DataFramesMeta.jl, `@eachrow` was named `@byrow`. 
+The old macro `@byrow` is deprecated, but the `@byrow` syntax can 
+now be used for similar, but not identical, behavior.
+
+To re-cap, the `@eachrow` rougly transforms
+
+```julia
+@eachrow df begin 
+    :a * :b
+end 
+```
+
+to 
+
+```julia
+begin
+    function tempfun(a, b)
+        for i in eachindex(a)
+            a[i] * b[i]
+        end
+    end
+    tempfun(df.a, df.b)
+    df
+end
+```
+
+The function `*` is applied by-row. But the result of those operations
+is not stored anywhere, as with `for`-loops in Base Julia. 
+Rather, `@eachrow` and `@eachrow!` return data frames. 
+
+Now consider `@byrow`. `@byrow` transforms
+
+```julia
+@with df @byrow begin 
+    :a * :b
+end
+```
+
+to 
+
+```julia
+tempfun(a, b) = a * b
+tempfun.(df.a, df.b)
+```
+
+In contrast to `@eachroe`, `@with` combined with `@byrow` will return a vector of the broadcasted multiplication and not a data frame.
+
+Additionally, `@eachrow` and `@eachrow!` allow modifying a data
+data frame. Just as with Base Julia broadcasting, `@byrow` will
+not update columns. 
+
+```julia
```

Comment:
How about making this a doctest?

Reply:
I will procrastinate this for a "jldoctest" PR.